opentelemetry-sdk/src/logs/log_processor.rs

```diff
@@ -115,10 +115,9 @@ impl LogProcessor for SimpleLogProcessor {
             });
         if let Err(err) = result {
             otel_error!(
-                name: "simple_log_processor_emit_error",
-                error = format!("{:?}", err)
+                name: "SimpleLogProcessor.Export.Error",
```

Comment:
its not easy to tell if this is useful or not.. The failure could be due to
1. Mutex poisoning
2. Exporter facing issues like network etc.
3. Futures-Executor can fail itself too.

I don't have a good solution in mind, so probably okay to just replace current global error handler with otel_error! for now until we can find a better alternative.

Reply:
Have modified it to differentiate between MutexPoisoning and Exporter failure - I think we can further extend it in future.